keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -237,9 +237,9 @@ class SqlArtifactRepository(
         .also { versions ->
           // FIXME: remove special handling for Docker
           return if (artifact is DockerArtifact) {
-            filterDockerVersions(artifact, versions)
+            filterDockerVersions(artifact, versions, limit)
           } else {
-            versions
+            versions.subList(0, Math.min(versions.size, limit))
```

Comment:
If we ever convert _all_ apps to using branch+timestamp for sorting, then this limit could be imposed in the SQL query, which would be much more efficient. Might be worth adding a comment to that effect here.

Reply:
In theory we could always assign a sort weight and persist that in its own column, making the query itself sortable.

Reply:
thanks @luispollo , good point - i'll add a comment for it. it's much better using `orderBy` or something similar and uses native SQL functions :)